### DISCUSSION TITLE
Text content rendered as unicode

### DISCUSSION THREAD

**Discussion by hilmanski** [discordjs/discord.js#8206] [Q&A] [answered]
Not really sure why.
when looping a messages on a channel

One of the text resulting in this
```
'Steam page is live! 🙂 \n' +
    '\n' +
    '🎉 🎉 \ud83c',
```

But should be (what's in discord)
```
Steam page is live! 🙂 

🎉 🎉 🎉
```

The unicode convert on the last emoji, make it hard for me to insert the data to database

Sample error when inserting to database (might be not relevant)
```
{
  message: 'invalid input syntax for type json',
  code: '22P02',
  details: 'Unicode low surrogate must follow a high surrogate.',
  hint: null
}
```
The main question is why it's converted the last convetti to that code

Thanks

**Comment by sinkaroid** [comment, marked as answer]
Does it's not automate converts in your database schema? and probably related to https://stackoverflow.com/questions/32158381/fatal-error-high-and-low-surrogate-code-points-are-not-valid-unicode-scalar-va  
  
In my case this issue happend to me as well, When parsing a texts with emoji, some emoji suddenly converted to unicode, but It is has been parsed correctly.

If this out from discord.js related, I think you should make a check statement before inserting your data to avoid this messy

**Reply by hilmanski** [reply]
Thanks. I think it's my mistake because cutting the string on certain length.

**Reply by kyranet** [reply]
~~If this issue is caused by you cutting a string, and not by writing strings manually (🎉 is `\uD83C\uDF89` btw), then you should use `[...string]` rather than a plain string.~~

~~The problem with `string.slice(0, N)` is that it *may* slice a codepoint, because `\uD83C` and `\uDF89` are two characters with the same codepoint, but when you do `[...string]`, `Hi 🎉` becomes `['H', 'i', ' ', '🎉']`, preserving the emoji fully.~~

**Edit**: nvm, I didn't read the issue correctly, I thought you were slicing it to fit the content in a Discord message, but it's a database slicing issue.
